lib/src/test/resources/compileContracts.sh

```diff
@@ -1,47 +1,134 @@
 #!/bin/bash
 
-pushd "$(dirname "$0")" || exit
-mkdir -p "compiled_v0"
+REPO_ROOT=$(dirname "$0")/../../../..
+
+V0_CONTRACT_PATH="src_v0"
+V1_CONTRACT_PATH="src_v1"
+V2_CONTRACT_PATH="src_v2"
+
+V0_ARTIFACT_PATH="compiled_v0"
+V1_ARTIFACT_PATH="compiled_v1"
+V2_ARTIFACT_PATH="compiled_v2"
+
+V1_COMPILER_VERSION="1.1.1"
+V2_COMPILER_VERSION="2.2.0"
+
+V1_COMPILER_BIN_PATH="compilers/v1"
+V2_COMPILER_BIN_PATH="compilers/v2"
+
+build_cairo_compilers() {
+  local VERSION=$1
+  local OUT_DIR=$2
+
+  local VERSION_SHORT=${VERSION:0:1}
+
+  if ! find "$REPO_ROOT/cairo$VERSION_SHORT"; then
+    echo "Cloning cairo repo..."
+    git clone -q https://github.com/starkware-libs/cairo.git "$REPO_ROOT/cairo$VERSION_SHORT"
+    pushd "$REPO_ROOT/cairo$VERSION_SHORT" || exit 1
+    git checkout -q "v$VERSION"
+    popd
+  else
+    echo "Found existing cairo repo, skipping cloning."
+  fi
```

Comment:
> I don't think we should cater to specific userbase, and go out of our way to ensure they have seemless experience when they want to develop.

I guess you are right, just that it was supported before and macos x64 seems like a more popular platform supporting which would be (mostly) effortless.

> we can allow to pass custom compiler to the script or something

Seems a bit complicated for a new user to understand, specifically considering it's not just one path and the correct file structure with corelib relative to compilers has to be maintained.  

> but I wouldn't include repo cloning and building the compiler just because someone someday may potentially try to use mac x86 to contribute.

I agree that it's kind of an overhead, although it's just `47M` (for reference, release is around `~107M`). It is true that building each compiler will take around `1GB`, but it was happening before and no one complained. 

But if you disagree with having it there, what do you think about including **"Supported platforms"** section in readme instead, specifying what version are officially supported, and moving instruction about git cloning there.

Reply:
Noone complained, because there was no other way 😢  now we at least have _some_ options, and are not forced to compile ourselves.

> what do you think about including "Supported platforms" section in readme instead

this is not a matter of supported platforms, because these do not change (it refers to the platforms you can run starknet-jvm on - and not platform you can contribute from). This is rather a topic of "you want to contribute? if you're on linux x86 or apple sillicon, just do it. Otherwise, you will need to compile the cairo compiler first".  So this is rather somethign that could go to contribution guide.

Reply:
> and are not forced to compile ourselves.

Because of CI we are 😅 And since we are anyway for now, in my vision it's not something hard to keep available to everyone. 

But I assume we're going forward with removing it here and moving it to contribution guidelines then as `"Platforms you can contribute from"`?

Reply:
For now - yes. As deiscussed here https://github.com/software-mansion/starknet-jvm/pull/285#discussion_r1324295333

>But I assume we're going forward with removing it here and moving it to contribution guidelines then as "Platforms you can contribute from"?

that's the beauty - you can contribute from any platform you want :+1: the only requirement is that if you're not using linux x64 or apple sillicon, you need to compile the compiler first, and that's basically it.